docker-compose.yml

```
@@ -0,0 +1,32 @@
+services:
+  integration_tests:
+    build:
+      dockerfile: Dockerfile
+      context: ${GSY_E_TARGET_REPO:-./}
+    container_name: ${CONTAINER_NAME:-gsy-e-tests}
+    entrypoint: ["gsy-e"]
+    command: ${COMMAND:-}
+    environment:
+      - REDIS_URL=${REDIS_URL:-redis://redis.container:6379/}
+    volumes:
+      - ${GSY_E_TARGET_REPO:-integration_tests}:/gsy-e
+
+  redis:
+    image: redis:6.2.5
+    restart: always
+    container_name: gsy-e-tests-redis
+    ports:
+      - "${PORT:-3679}:${PORT:-6379}"
+    hostname: ${HOST:-redis.container}
+    volumes:
+      - redis_volume:/data
+
+volumes:
+  integration_tests:
+    driver: local
+  redis_volume:
+    driver: local
```

Why is `volumes` declared at top-level? And could you explain why we need to specify the driver (and why `local` specifically)?

---

> Why is volumes declared at top-level?

This is in order to create the docker volume which is used for a given service (apart from the mount type).


> And could you explain why we need to specify the driver (and why local specifically)?

AFAIU, think this is not strictly necessary. The local driver is the default value, so this is how would the default configuration should look like, if made explicit.

I originally did it like this in order to be "mentally" aware of how the values were set. 

This can be safely rewritten as follows, which I will update in a future commit.
```suggestion
volumes:
  integration_tests:
  redis_volume:
```

---

> This can be safely rewritten as follows, which I will update in a future commit.

What's the current progress on this?

---

Adapted on GBIT's compose files.